<h2><font color="#004D7F" size=6>Module 2. Data Analysis</font></h2>

<h1><font color="#004D7F" size=5>2. Descriptive Statistics</font></h1>

<h2><font color="#004D7F" size=5>Index</font></h2>
<a id="indice"></a>

* 1. Introduction
    * 1.1. Load the Dataset
* 2. Descriptive Statistics Functions
    * 2.1. Reviewing the Data: head()
    * 2.2. Data Dimensions: shape
    * 2.3. Data Types: dtypes
    * 2.4. Summary: describe()
    * 2.5. Class Distribution: groupby('class').size()
    * 2.6. Correlations: corr()
    * 2.7. Skewness: skew()

# <font color="#004D7F"> 1. Introduction</font>

One of the main mistakes made when starting to work with machine learning is making decisions directly through the algorithms without prior analysis of the dataset to work with. It is important to understand that, data analysis and good preprocessing of the data before performing modeling, will not only lead us to obtain better and more reliable results, but will also help us understand our dataset perfectly, providing a great experience over time in the analysis and data processing phase.

## <font color="#004D7F">1.1. Load the Dataset</font>

For this practice, we are going to load the Pima Indian Diabetes dataset to make observations with descriptive statistics functions.

In [2]:
import pandas as pd

filename = 'data/pima-indians-diabetes.csv'
names = ['preg','plas','pres','skin','test','mass','pedi','age','class']

data = pd.read_csv(filename, names=names)
data

preg  plas  pres  skin  test  mass     pedi  age  class
0       6   148    72    35     0  33.6   627.00   50      1
1       1    85    66    29     0  26.6   351.00   31      0
2       8   183    64     0     0  23.3   672.00   32      1
3       1    89    66    23    94  28.1   167.00   21      0
4       0   137    40    35   168  43.1  2288.00   33      1
..    ...   ...   ...   ...   ...   ...      ...  ...    ...
763    10   101    76    48   180  32.9   171.00   63      0
764     2   122    70    27     0  36.8     0.34   27      0
765     5   121    72    23   112  26.2   245.00   30      0
766     1   126    60     0     0  30.1   349.00   47      1
767     1    93    70    31     0  30.4   315.00   23      0

[768 rows x 9 columns]

## <font color="#004D7F">2.1. Reviewing the data: head() </font>

You can review the first 20 rows of your data using the head() function in the Pandas DataFrame. You can see that the first column lists the row number, which is useful for referencing a specific observation.

In [3]:
data.head(20)

preg  plas  pres  skin  test  mass    pedi  age  class
0      6   148    72    35     0  33.6   627.0   50      1
1      1    85    66    29     0  26.6   351.0   31      0
2      8   183    64     0     0  23.3   672.0   32      1
3      1    89    66    23    94  28.1   167.0   21      0
4      0   137    40    35   168  43.1  2288.0   33      1
5      5   116    74     0     0  25.6   201.0   30      0
6      3    78    50    32    88  31.0   248.0   26      1
7     10   115     0     0     0  35.3   134.0   29      0
8      2   197    70    45   543  30.5   158.0   53      1
9      8   125    96     0     0   0.0   232.0   54      1
10     4   110    92     0     0  37.6   191.0   30      0
11    10   168    74     0     0  38.0   537.0   34      1
12    10   139    80     0     0  27.1  1441.0   57      0
13     1   189    60    23   846  30.1   398.0   59      1
14     5   166    72    19   175  25.8   587.0   51      1
15     7   100     0     0     0  30.0   484.0   32      1
16     0   118    84    47   230  45.8   551.0   31      1
17     7   107    74     0     0  29.6   254.0   31      1
18     1   103    30    38    83  43.3   183.0   33      0
19     1   115    70    30    96  34.6   529.0   32      1

## <font color="#004D7F">2.2. Dimensiones de los datos: shape </font>

Puede revisar la forma y el tamaño de su conjunto de datos imprimiendo la propiedad shape en el DataFrame de Pandas. Los resultados se enumeran en filas y luego en columnas. Puede ver que el conjunto de datos tiene 768 filas y 9 columnas.

In [5]:
# Dimensions of your data
data.shape

(768, 9)